packages/dds/matrix/src/matrix.ts

```diff
@@ -101,14 +156,30 @@ export class SharedMatrix<T = any>
 
 	private cells = new SparseArray2D<MatrixItem<T>>(); // Stores cell values.
 	private pending = new SparseArray2D<number>(); // Tracks pending writes.
+	private cellLastWriteTracker = new SparseArray2D<CellLastWriteTrackerItem>(); // Tracks last writes sequence numner in a cell.
+	// Tracks the seq number of Op at which policy switch happens from Last Write Win to First Write Win.
+	private setCellPolicySwitchOpSeqNumber = -1;
 
+	/**
+	 * Constructor for the Shared Matrix
+	 * @param runtime - DataStore runtime.
+	 * @param id - id of the dds
+	 * @param attributes - channel attributes
+	 * @param _isSetCellConflictResolutionPolicyFWW - Conflict resolution for Matrix set op is First Writer Win in case of
+	 * race condition. Client can still overwrite values in case of no race.
+	 */
 	constructor(
 		runtime: IFluidDataStoreRuntime,
 		public id: string,
 		attributes: IChannelAttributes,
+		_isSetCellConflictResolutionPolicyFWW?: boolean,
 	) {
 		super(id, runtime, attributes, "fluid_matrix_");
 
+		if (_isSetCellConflictResolutionPolicyFWW === true) {
+			this.setCellPolicySwitchOpSeqNumber = 0;
```

Comment:
nit: I'd move all initialization paths here:
this.setCellPolicySwitchOpSeqNumber = _isSetCellConflictResolutionPolicyFWW ? 0 : -1;